caido: init at 0.29.2

octodi opened the pull request:
## Description of changes

Added caido a [A lightweight web security auditing toolkit.](https://caido.io/)

https://github.com/caido/caido

Related to https://github.com/NixOS/nixpkgs/issues/81418

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

arthsmn commented:
Your commit should follow [the commit convention](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions). You just need to use the title of the PR as the commit message.

arthsmn commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,39 @@
+  src = fetchurl {
+    url = "https://storage.googleapis.com/caido-releases/v${version}/${name}.AppImage";
+    name = "caido-${version}.AppImage";
+    sha256 = "sha256-4PgQK52LAX1zacmoUK0muIhrvFDF7anQ6sx35I+ErVs=";
```
```suggestion
    hash = "sha256-4PgQK52LAX1zacmoUK0muIhrvFDF7anQ6sx35I+ErVs=";
```
Use hash instead of sha256.

arthsmn commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib, fetchurl, appimageTools, makeWrapper }:
```
```suggestion
{ lib
, fetchurl
, appimageTools
, makeWrapper
}:
```
This fits better the style of nixpkgs.

arthsmn commented:
You should also squash the commits.

arthsmn commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,43 @@
+  pname = "caido";
+  version = "0.29.2";
+  name = "caido-desktop-linux-v${version}-e0f8102b";
+  nameExecutable = pname;
```
Is this really necessary?

octodi replied:
`name` or `nameExecutable`?

arthsmn replied:
nameExecutable

octodi replied:
Does it now look [good](https://github.com/NixOS/nixpkgs/pull/270664/commits/333185fdb8fb471d11b0f0ad71b35477a07399b2)?

arthsmn replied:
It looks better. You still need to squash the commits.

octodi replied:
[Done](https://github.com/NixOS/nixpkgs/pull/270664/commits/7c6b4185db3d7afecae0aba7d0d409c336e84178)

arthsmn commented:
I don't know if this is due to your packaging or the program itself, but on plasma5 wayland when I open the binary it stays like this:
![Screenshot_20231128_151303](https://github.com/NixOS/nixpkgs/assets/150680976/35d1a840-604a-4301-8c5c-8a7801920720)
It gives me these errors on the terminal:
```
Gtk-Message: 15:12:50.426: Failed to load module "colorreload-gtk-module"
Gtk-Message: 15:12:50.427: Failed to load module "window-decorations-gtk-module"
Could not determine the accessibility bus address
Gtk-Message: 15:12:50.516: Failed to load module "colorreload-gtk-module"
Gtk-Message: 15:12:50.516: Failed to load module "window-decorations-gtk-module"
Cannot get default EGL display: EGL_BAD_PARAMETER
Cannot create EGL sharing context: invalid display (last error: EGL_SUCCESS)

(WebKitWebProcess:225599): Gdk-ERROR **: 15:12:50.562: The program 'WebKitWebProcess' received an X Window System error.
This probably reflects a bug in the program.
The error was '167'.
  (Details: serial 178 error_code 167 request_code 150 (unknown) minor_code 21)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```

arthsmn commented:
> > I don't know if this is due to your packaging or the program itself, but on plasma5 wayland when I open the binary it stays like this: ![Screenshot_20231128_151303](https://private-user-images.githubusercontent.com/150680976/286359861-35d1a840-604a-4301-8c5c-8a7801920720.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNDA3MjUsIm5iZiI6MTcwMTI0MDQyNSwicGF0aCI6Ii8xNTA2ODA5NzYvMjg2MzU5ODYxLTM1ZDFhODQwLTYwNGEtNDMwMS04YzVjLThhNzgwMTkyMDcyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOVQwNjQ3MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZDNhYWIyYjE2Y2ViZTYzMTYwYzRiMDY0NWM1MzFlMTllYTcxNGI0NzEwOWViNTY4OTgwNTk5MTIyZjFhNzgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.L-uGqKxz_ztpwMfQbSxhnUFjMpXIO4yMVPJ6dpe4JTw) It gives me these errors on the terminal:
> > ```
> > Gtk-Message: 15:12:50.426: Failed to load module "colorreload-gtk-module"
> > Gtk-Message: 15:12:50.427: Failed to load module "window-decorations-gtk-module"
> > Could not determine the accessibility bus address
> > Gtk-Message: 15:12:50.516: Failed to load module "colorreload-gtk-module"
> > Gtk-Message: 15:12:50.516: Failed to load module "window-decorations-gtk-module"
> > Cannot get default EGL display: EGL_BAD_PARAMETER
> > Cannot create EGL sharing context: invalid display (last error: EGL_SUCCESS)
> > 
> > (WebKitWebProcess:225599): Gdk-ERROR **: 15:12:50.562: The program 'WebKitWebProcess' received an X Window System error.
> > This probably reflects a bug in the program.
> > The error was '167'.
> >   (Details: serial 178 error_code 167 request_code 150 (unknown) minor_code 21)
> >   (Note to programmers: normally, X errors are reported asynchronously;
> >    that is, you will receive the error a while after causing it.
> >    To debug your program, run it with the GDK_SYNCHRONIZE environment
> >    variable to change this behavior. You can then get a meaningful
> >    backtrace from your debugger if you break on the gdk_x_error() function.)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also I noticed that even though you're setting the platform-hint to auto the app opens on a xwayland window.
> 
>     * But it worked for me, try reopening once
>       ![Screenshot from 2023-11-29 12-16-35](https://private-user-images.githubusercontent.com/127038896/286502005-4cfcdbca-a063-4b91-9441-f9b48861c621.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNjIyNDgsIm5iZiI6MTcwMTI2MTk0OCwicGF0aCI6Ii8xMjcwMzg4OTYvMjg2NTAyMDA1LTRjZmNkYmNhLWEwNjMtNGI5MS05NDQxLWY5YjQ4ODYxYzYyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOVQxMjQ1NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNGRiYTFiNDZiZTM0ZjY4NGM2NDZhMjQ5NzhiNjExYTU2ZTU4ODVkN2Y3YzVkYzQ4NzQ2NmRiNzE0MDRlNDA5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XL4i6EDhI76XphjoqdKJwkv-l_TLzY8VCuubi8TngYA)
> 
> 
> OR try this config
> 
> ```nix
> { lib
> , fetchurl
> , appimageTools
> }:
> 
> let
>   pname = "caido";
>   version = "0.29.2";
>   src = fetchurl {
>     url = "https://storage.googleapis.com/caido-releases/v${version}/caido-desktop-linux-v${version}-e0f8102b.AppImage";
>     hash = "sha256-4PgQK52LAX1zacmoUK0muIhrvFDF7anQ6sx35I+ErVs=";
>   };
>   appimageContents = appimageTools.extractType2 { inherit pname src version; };
> 
> in appimageTools.wrapType2 {
>   inherit pname src version;
> 
>   extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs) ++ [ pkgs.libthai ];
> 
>   extraInstallCommands = ''
>     mv $out/bin/${pname}-${version} $out/bin/${pname}
>     install -m 444 -D ${appimageContents}/caido.desktop -t $out/share/applications
>     substituteInPlace $out/share/applications/caido.desktop \
>       --replace 'Exec=AppRun' 'Exec=${pname}'
>     install -m 444 -D ${appimageContents}/caido.png \
>       $out/share/icons/hicolor/512x512/apps/caido.png
>   '';
> 
>   meta = with lib; {
>     description = "A lightweight web security auditing toolkit.";
>     homepage = "https://caido.io/";
>     license = licenses.unfree;
>     maintainers = with maintainers; [ octodi ];
>     platforms = [ "x86_64-linux" ];
>   };
> }
> ```

I tested on a VM, altough it works it is still not using wayland.

octodi commented:
> > > I don't know if this is due to your packaging or the program itself, but on plasma5 wayland when I open the binary it stays like this: ![Screenshot_20231128_151303](https://private-user-images.githubusercontent.com/150680976/286359861-35d1a840-604a-4301-8c5c-8a7801920720.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNDA3MjUsIm5iZiI6MTcwMTI0MDQyNSwicGF0aCI6Ii8xNTA2ODA5NzYvMjg2MzU5ODYxLTM1ZDFhODQwLTYwNGEtNDMwMS04YzVjLThhNzgwMTkyMDcyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOVQwNjQ3MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZDNhYWIyYjE2Y2ViZTYzMTYwYzRiMDY0NWM1MzFlMTllYTcxNGI0NzEwOWViNTY4OTgwNTk5MTIyZjFhNzgzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.L-uGqKxz_ztpwMfQbSxhnUFjMpXIO4yMVPJ6dpe4JTw) It gives me these errors on the terminal:
> > > ```
> > > Gtk-Message: 15:12:50.426: Failed to load module "colorreload-gtk-module"
> > > Gtk-Message: 15:12:50.427: Failed to load module "window-decorations-gtk-module"
> > > Could not determine the accessibility bus address
> > > Gtk-Message: 15:12:50.516: Failed to load module "colorreload-gtk-module"
> > > Gtk-Message: 15:12:50.516: Failed to load module "window-decorations-gtk-module"
> > > Cannot get default EGL display: EGL_BAD_PARAMETER
> > > Cannot create EGL sharing context: invalid display (last error: EGL_SUCCESS)
> > > 
> > > (WebKitWebProcess:225599): Gdk-ERROR **: 15:12:50.562: The program 'WebKitWebProcess' received an X Window System error.
> > > This probably reflects a bug in the program.
> > > The error was '167'.
> > >   (Details: serial 178 error_code 167 request_code 150 (unknown) minor_code 21)
> > >   (Note to programmers: normally, X errors are reported asynchronously;
> > >    that is, you will receive the error a while after causing it.
> > >    To debug your program, run it with the GDK_SYNCHRONIZE environment
> > >    variable to change this behavior. You can then get a meaningful
> > >    backtrace from your debugger if you break on the gdk_x_error() function.)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Also I noticed that even though you're setting the platform-hint to auto the app opens on a xwayland window.
> > 
> > 
> > ```
> > * But it worked for me, try reopening once
> >   ![Screenshot from 2023-11-29 12-16-35](https://private-user-images.githubusercontent.com/127038896/286502005-4cfcdbca-a063-4b91-9441-f9b48861c621.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNjIyNDgsIm5iZiI6MTcwMTI2MTk0OCwicGF0aCI6Ii8xMjcwMzg4OTYvMjg2NTAyMDA1LTRjZmNkYmNhLWEwNjMtNGI5MS05NDQxLWY5YjQ4ODYxYzYyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOVQxMjQ1NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNGRiYTFiNDZiZTM0ZjY4NGM2NDZhMjQ5NzhiNjExYTU2ZTU4ODVkN2Y3YzVkYzQ4NzQ2NmRiNzE0MDRlNDA5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XL4i6EDhI76XphjoqdKJwkv-l_TLzY8VCuubi8TngYA)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > OR try this config
> > ```nix
> > { lib
> > , fetchurl
> > , appimageTools
> > }:
> > 
> > let
> >   pname = "caido";
> >   version = "0.29.2";
> >   src = fetchurl {
> >     url = "https://storage.googleapis.com/caido-releases/v${version}/caido-desktop-linux-v${version}-e0f8102b.AppImage";
> >     hash = "sha256-4PgQK52LAX1zacmoUK0muIhrvFDF7anQ6sx35I+ErVs=";
> >   };
> >   appimageContents = appimageTools.extractType2 { inherit pname src version; };
> > 
> > in appimageTools.wrapType2 {
> >   inherit pname src version;
> > 
> >   extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs) ++ [ pkgs.libthai ];
> > 
> >   extraInstallCommands = ''
> >     mv $out/bin/${pname}-${version} $out/bin/${pname}
> >     install -m 444 -D ${appimageContents}/caido.desktop -t $out/share/applications
> >     substituteInPlace $out/share/applications/caido.desktop \
> >       --replace 'Exec=AppRun' 'Exec=${pname}'
> >     install -m 444 -D ${appimageContents}/caido.png \
> >       $out/share/icons/hicolor/512x512/apps/caido.png
> >   '';
> > 
> >   meta = with lib; {
> >     description = "A lightweight web security auditing toolkit.";
> >     homepage = "https://caido.io/";
> >     license = licenses.unfree;
> >     maintainers = with maintainers; [ octodi ];
> >     platforms = [ "x86_64-linux" ];
> >   };
> > }
> > ```
> 
> I tested on a VM, altough it works it is still not using wayland.

I unable to launch it as a native wayland window suggest me a config if you know.

octodi commented:
Wrapping it with `ozone-platform=wayland` , also didn't work

```nix
wrapProgram $out/bin/${pname} \
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform=wayland --enable-features=WaylandWindowDecorations}}"
```

arthsmn commented:
I know why it isn't working. It uses [tauri](https://tauri.app/), not electron. I'm trying to understand if it has (proper) wayland support.

arthsmn commented:
I solved my issue using this environment variable: `WEBKIT_DISABLE_COMPOSITING_MODE=1`

But the app stills default to X.

arthsmn commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,40 @@
+  extraInstallCommands = ''
+    mv $out/bin/${pname}-${version} $out/bin/${pname}
+    source "${makeWrapper}/nix-support/setup-hook"
+    wrapProgram $out/bin/${pname} \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```
```suggestion
```
It is of no use as it is not using electron.

octodi replied:
[Done](https://github.com/NixOS/nixpkgs/pull/270664/commits/2fd5622d83294dd257f65bda4fcbb9460e423c13)

arthsmn reviewed (approved):
Using it through xwayland should work just fine. As wayland support seems to be missing upstream this package is just fine.

fabaff commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,36 @@
+  '';
+
+  meta = with lib; {
+    description = "A lightweight web security auditing toolkit.";
```
No period at the end please.

```suggestion
    description = "A lightweight web security auditing toolkit";
```

fabaff commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,36 @@
+
+  meta = with lib; {
+    description = "A lightweight web security auditing toolkit.";
+    homepage = "https://caido.io/";
```
Tool (e. g., `nix-update`) will take the `changelog` content and add the details to commit message and/or the PR description.

```suggestion
    homepage = "https://caido.io/";
    changelog = "https://github.com/caido/caido/releases/tag/v${version}";
```

pbsds reviewed (commented):
Result of `nixpkgs-review pr 270664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>caido</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

---

Works with `WEBKIT_DISABLE_COMPOSITING_MODE=1` over x forwarding, but is blank without. Should we wrap the program with this set?

arthsmn commented:
> Result of `nixpkgs-review pr 270664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package built:
> 
>     * [x]  package path fits guidelines
> 
>     * [x]  package name fits guidelines
> 
>     * [x]  package version fits guidelines
> 
>     * [x]  package builds on x86_64-linux
> 
>     * [x]  executables tested on x86_64-linux
> 
>     * [x]  `meta.description` is set and fits guidelines
> 
>     * [x]  `meta.license` fits upstream license
> 
>     * [x]  `meta.platforms` is set
> 
>     * [x]  `meta.maintainers` is set
> 
>     * [x]  build time only dependencies are declared in `nativeBuildInputs`
> 
>     * [x]  source is fetched using the appropriate function
> 
>     * [x]  the list of `phases` is not overridden
> 
>     * [x]  when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
> 
>     * [x]  patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
> 
>     * [x]  patches that are remotely available are fetched rather than vendored
> 
> 
> Works with `WEBKIT_DISABLE_COMPOSITING_MODE=1` over x forwarding, but is blank without. Should we wrap the program with this set?

It seems to be a specific problem that target certain environments, I don't know if it's safe and worth to use this flag by default.

pbsds commented:
We experience the issue on plasma, gnome and over x11 forwarding. A workaround, albeit temporary, is warranted.

This package won't be the first:

```
$ rg WEBKIT_DISABLE_COMPOSITING_MODE -B2
nixpkgs/pkgs/applications/audio/soundux/default.nix
131-    # Work around upstream bug https://github.com/Soundux/Soundux/issues/435
132-    wrapProgram "$out/bin/soundux" \
133:      --set WEBKIT_DISABLE_COMPOSITING_MODE 1 \

nixpkgs/pkgs/applications/misc/ulauncher/default.nix
107-     --prefix PATH : "${lib.makeBinPath [ wmctrl ]}"
108-     --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libX11 ]}"
109:     --prefix WEBKIT_DISABLE_COMPOSITING_MODE : "1"

nixpkgs/pkgs/applications/misc/holochain-launcher/default.nix
50-    gappsWrapperArgs+=(
51-      --prefix XDG_DATA_DIRS : "${shared-mime-info}/share"
52:      --set WEBKIT_DISABLE_COMPOSITING_MODE 1

nixpkgs/pkgs/tools/networking/gp-saml-gui/default.nix
34-  preFixup = ''
35-    gappsWrapperArgs+=(
36:      --set WEBKIT_DISABLE_COMPOSITING_MODE "1"
```



Please make an issue upstream explaining the issue and link to it in a comment above the workaround. Upstream might be interested in a possible fix is described in https://github.com/Ulauncher/Ulauncher/issues/114#issuecomment-1099714353 ;)

pbsds commented:
It seems popular projects like dorion [expose it as an option](
https://github.com/SpikeHD/Dorion/blob/41dba036c99fadd53ced7e056ab6a1f9383f2d95/src-tauri/src/functionality/window.rs#L80)

octodi committed:
caido: init at 0.29.2

Removed OZONE flags as it not an electron app

Added changelog

Added WEBKIT_DISABLE_COMPOSITING_MODE env variable

Added mainProgram and sourceProvenance

Co-Authored-By: Fabian Affolter <mail@fabian-affolter.ch>
Co-Authored-By: Arthur <150680976+arthsmn@users.noreply.github.com>
Co-Authored-By: Peder Bergebakken Sundt <pbsds@hotmail.com>

arthsmn commented:
Revert to the first commit and force-push.

arthsmn commented:
Also, you should edit the commit message to remove unnecessary stuff.

octodi commented:
> Also, you should edit the commit message to remove unnecessary stuff.

[Done](https://github.com/NixOS/nixpkgs/pull/270664/commits/2e36b2f1016f4ab13dfd98143fb1d9c5dd7947a0)

pbsds commented on pkgs/by-name/ca/caido/package.nix:
```diff
@@ -0,0 +1,41 @@
+    homepage = "https://caido.io/";
+    changelog = "https://github.com/caido/caido/releases/tag/v${version}";
+    license = licenses.unfree;
+    maintainers = with maintainers; [ octodi ];
```
```suggestion
    maintainers = with maintainers; [ octodi ];
    mainProgram = "caido";
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

pbsds commented:
Result of `nixpkgs-review pr 270664` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>caido</li>
  </ul>
</details>
